### ISSUE TITLE
The new installer didn't grant the correct permission.

### ISSUE THREAD

**Issue by snowie2000** [snowie2000/mactype#470]
The new installer only granted full control for Administrators to the application folder. But for some reason, the inheritance functionality doesn't work properly.

If you check all the files and folders installed with the installer, they all have the right permission. But if you create or copy files from other folders to it, they won't get the right permission. 
By right-clicking the newly added files, you can see there are shields near the `delete` and `rename` commands while the files already installed don't have. 
Plus, when you want to overwrite or delete files from the application folder, you get a UAC warning which is also a consequence of the incorrect folder permission.
With that set, MacType will not able to switch and read profiles unless it is running with the Administrator privilege.

**Comment by sammilucia** [comment]
Thanks @snowie2000, investigating

**Comment by sammilucia** [comment]
@snowie2000 just a note - it's best to not have Full Control on a root folder inside Program Files, because typically we want UAC to warn the user if exe or dll files are about to be modified by a malicious process.

Best practice would be to set the \ini folder only as Full Control. However MacType.ini is in the root folder so we can't.

Possible solutions:
1. Make whole folder Full Control for Users group - don't follow best practice
2. Make just MacType.ini Read/Write for Users group
3. Make \ini Read/Write for Users group - and have MacWiz.exe look inside \ini for MacType.ini

**Comment by snowie2000** [comment]
It's interesting that sometimes, although a file has the right permission (at least it looks like so), I can read/write the file, but not the program.
Grant full control to a root folder is indeed a lazy and insecure solution, but it works for me. 😄

Okay, put mactype.ini into the ini folder is not very easy, since every profile has the same extension ".ini", so additional codes are required to distinguish mactype.ini from others.
Is there a way to allow non-privileged process to have full control of the mactype.ini file?

**Comment by snowie2000** [comment]
Good. Download and testing.

**Comment by snowie2000** [comment]
So far so good, permissions seem correct.
Why you remove the logic of cleaning up the AppInit_DLLs keys?

**Comment by snowie2000** [comment]
One more problem:
Looks like the installer didn't create any entries in the start menu. Is it what it designed to be?

**Comment by sammilucia** [comment]
Dammit, nope! I must have introduced a regression. Looking...

**Comment by sammilucia** [comment]
Regarding AppInit_DLLs ... because MacWiz.exe handles this anyway doesn't it? I thought there's no point recreating logic to handle it in the installer if MacWiz handles it anyway... Let me know though... 😜

**Comment by snowie2000** [comment]
Not in all situations.
If a user uninstalled MacType (with registry mode currently enabled), there is no chance for MacWiz to clean up the keys. Once the user reinstalls MacType, he/she will be in registry mode once the installation is complete.

**Comment by sammilucia** [comment]
Hmm, okay. I've put back the logic and made it more robust. But I just realised there's one use case where it will wipe AppInit_DLLs when we don't want it to - if you re-run the installer or do an in-place upgrade. Need to add a check for that

CHANGES:
- Made AppInit_DLLs cleanup more robust
- Fixed Desktop icons not being created under some circumstances
- Fixed Start Menu Icons not being created under some circumstances
- Added option for minimal install

TODO:
- Needs fixing: If setup is run again (without uninstalling) or in place upgrade, it will wipe AppInit_DLLs

**Comment by sammilucia** [comment]
r6 of new installer. All issues should be resolved but needs more testing - I've added new checks and logic. @snowie2000 can you please test and if okay let me know if you want to package another beta MacType + installer beta before stable release?

CHANGES
- Added logic to only clear AppInit_DLLs the first time the new installer is run
- Added flag to disable selecting install folder if MacType already installed by new installer
- Added flag to disable selecting Start Menu folder if MacType already installed by new installer
- Turned on version checking for all files except ini and txt files
- (Fixed regression) Re-added LoadType=1 (see TODO below)

TODO
- @snowie2000 after a fresh install, If you run MacWiz.exe and select Registry Mode, MacWiz.exe reports mode hasn't changed and AppInit_DLLs isn't written - will log a new issue

**Comment by sammilucia** [comment]
Closing as resolved
